services/hdbpp-cluster-reporting/server/routes/attributes.py

```diff
@@ -0,0 +1,126 @@
+
+def get_from_db(att_format, att_type, req):
+    """
+    Generic function that just grabs the data from the database according
+    to the format and type.
+    Attributes:
+    -----------
+        att_format: str -- Format of the attribute to query.
+        att_type: str -- Type of the attribute to query.
+        req: SQLAlchemy InstrumentedAttributes -- Output field requested.
+    """
+    if att_type not in config.DB_TYPES or att_format not in config.DB_FORMAT:
+        raise Exception("The type: {} and format: {} is not supported.".format(att_type, att_format))
+
+    attributes_result = models.Attributes.query \
+            .with_entities(req.label('result')) \
+            .filter(models.Attributes.att_type == att_type, models.Attributes.att_format == att_format)
+    return jsonify(attributes_result[0].result)
+
+class AttributesCount(Resource):
+    def get(self):
+        attributes_result = models.Attributes.query.with_entities(func.sum(models.Attributes.att_count).label('c'))
+        return jsonify(attributes_result[0].c)
+
+class AttributesTypeOrFormatCount(Resource):
+    def get(self, att_info):
+        if att_info not in config.DB_TYPES and att_info not in config.DB_FORMAT:
+            raise Exception("The type or format: {} is not supported.".format(att_info))
+        
+        attributes_result = models.Attributes.query.with_entities(models.Attributes.att_type) \
+                .filter(models.Attributes.att_type == att_info)
+        
+        #Check if the information passed is a type, if not it should be a format.
+        if attributes_result.count() == 0:
+            attributes_result = models.Attributes.query.with_entities(func.sum(models.Attributes.att_count).label('c')) \
+                    .group_by(models.Attributes.att_format).filter(models.Attributes.att_format == att_info)
+        else:
+            attributes_result = models.Attributes.query.with_entities(func.sum(models.Attributes.att_count).label('c')) \
+                    .group_by(models.Attributes.att_type).filter(models.Attributes.att_type == att_info)
+        
+        if attributes_result.count() == 0:
+            return 0
+        
+        return jsonify(attributes_result[0].c)
+
+class AttributesFormat(Resource):
+    def get(self):
+        return jsonify(config.DB_FORMAT)
+
+class AttributesType(Resource):
+    def get(self):
+        return jsonify(config.DB_TYPES)
+
+class AttributesFormatCount(Resource):
+    def get(self, att_format):
+        attributes_result = models.Attributes.query \
+                .with_entities(func.sum(models.Attributes.att_count).label('c')) \
+                .group_by(models.Attributes.att_format) \
+                .filter(models.Attributes.att_format == att_format)
+        return jsonify(attributes_result[0].c)
+
+class AttributesFormatTypeCount(Resource):
+    def get(self, att_format, att_type):
+        return get_from_db(att_format, att_type, models.Attributes.att_count)
```

Comment:
We need to handle the exception from get_from_db somehow, and return a message/code.